mesonbuild/compilers/c.py

```diff
@@ -1027,4 +1027,28 @@ def get_instruction_set_args(self, instruction_set):
             return None
         return vs32_instruction_set_args.get(instruction_set, None)
 
+    def get_toolset_version(self):
+        # See boost/config/compiler/visualc.cpp for up to date mapping
+        version = int(''.join(self.version.split('.')[0:2]))
```

Comment:
This should be wrapped in a `try:` statement with an error if the version is unparseable. Can boost versions ever contain non-numeral characters? If so, this should use a sensible fallback instead.

Reply:
This is actually the mapping between the Visual Studio version like (14.0 or 14.1) and the compiler define _MSC_VER like 1900 and 1910. So it isn't boost related except that this is "experimentally found" by users of Visual Studio. There is no official mapping from MS. So it should always be numeric. I will add a try: block.

Reply:
Ah, makes sense.